homeassistant/components/contec_controllers/light.py

```diff
@@ -0,0 +1,76 @@
+"""Contec light entity."""
+
+import logging
+from typing import List
+
+from ContecControllers.ContecOnOffActivation import ContecOnOffActivation
+from ContecControllers.ControllerManager import ControllerManager
+
+from homeassistant.components.light import LightEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+):
+    """Set up the Contec Lights."""
+    controllerManager: ControllerManager = hass.data[DOMAIN][config_entry.entry_id]
+    allLights: List[ContecLight] = []
+    for onOff in controllerManager.OnOffActivations:
+        allLights.append(ContecLight(onOff))
+    async_add_entities(allLights)
+
+
+class ContecLight(LightEntity):
+    """Representation of a Contec light."""
+
+    _id: str
+    _name: str
+    _onOffActivation: ContecOnOffActivation
+
+    def __init__(self, onOffActivation: ContecOnOffActivation):
+        """Initialize an ContecLight."""
+        self._onOffActivation = onOffActivation
+        self._id = f"light_{onOffActivation.ControllerUnit.UnitId}-{onOffActivation.StartActivationNumber}"
+        self._name = f"Contec Light {self._id}"
+
+        def StateUpdated(isOn: bool):
+            self.schedule_update_ha_state()
```

Comment:
```suggestion
            self.async_write_ha_state()
```

If its async you can use `async_write_ha_state`